thoth/package_extract/image.py

```diff
@@ -546,6 +546,47 @@ def get_image_size(path: str) -> int:
     return total_size
 
 
+def _get_cuda_version(path: str) -> dict:
+    """Get the cuda version."""
+    res = dict()
+    # gathering version from version.txt file
+    version_path = path + "/usr/local/cuda/version.txt"
+    if os.path.isfile(version_path):
+        with open(version_path, "r") as f:
+            output = f.read()
+            lines = output.split()
```

Comment:
Here, we split it and `lines = ['CUDA', 'Version', '10.1.243']`.

Reply:
What about reading the file line by line and check if a line does not start with `CUDA Version` string?

```python
for line in f.readline():
    if line.startswith("CUDA Version"):
        cuda_version = line[len("CUDA Version"):]
        break
else:
     _LOGGER.warning("No CUDA version identifier was found in file %r", version_path[root_path:])
```